cookbooks/bcpc/recipes/keepalived.rb

```diff
@@ -20,7 +20,23 @@
 include_recipe "bcpc::default"
 
 make_config('keepalived-router-id', "#{(rand * 1000).to_i%254/2*2+1}")
-make_config('keepalived-password', secure_password)
+
+keepalived_password = get_config('keepalived-password')
+if keepalived_password.nil?
+  keepalived_password = secure_password
+end
+
+bootstrap = get_all_nodes.select{|s| s.hostname.include? 'bootstrap'}[0].fqdn
+
+nodes = get_nodes_for("keepalived").map!{ |x| x['fqdn'] }.join(",")
```

Comment:
Instead of doing a `map!` on `get_nodes_for`, please use the node attribute function [`get_node_attributes`](https://github.com/bloomberg/chef-bach/blob/94b5c20e7ed967e111daae6c05d15cb5642e381a/cookbooks/bcpc/libraries/utils.rb#L176-L180).